src/array.rs

```diff
@@ -320,36 +321,50 @@ where
     T: ArrayValue,
 {
     let mut chunk_h3_map = HashMap::<&T, CellCoverage>::default();
-    let window_box = h3o::geom::Rect::from_degrees(window_box)?;
-    for cell in window_box.to_cells(
-        PolyfillConfig::new(h3_resolution).containment_mode(ContainmentMode::ContainsCentroid),
-    ) {
-        // find the array element for the coordinate of the h3 index
-        let coord: Coord = LatLng::from(cell).into();
-        let arr_coord = {
-            let transformed = inverse_transform * coord;
-
-            match axis_order {
-                AxisOrder::XY => [
-                    transformed.x.floor() as usize,
-                    transformed.y.floor() as usize,
-                ],
-                AxisOrder::YX => [
-                    transformed.y.floor() as usize,
-                    transformed.x.floor() as usize,
-                ],
-            }
-        };
-        if let Some(value) = arr.get(arr_coord) {
-            if let Some(nodata) = nodata_value {
-                if nodata == value {
-                    continue;
+
+    for splitted_window_box in split_rect_at_antimeridian(window_box) {
+        // h3 is only defined within -180 ... 180, so all boxes after the antimeridian split should be
+        // in this range.
```

Comment:
Indeed, and I should probably check this when a new geometry is created!

Ideally it should even be checked at the level of the `LatLng` type, but internally it sometimes goes out of this range so it's a bit more complex to pull off.

Reply:
I would not worry too much about checking this within h3o. It would be worth mentioning it in the docs - maybe on the `LatLng`  type.

I suppose tackling it a type-safe way would require having a second LatLng-like type for internal use. That one then could allow going out of this range.